bench: refactor random number generation in `stats/base/dists/signrank`

shraddha761 opened the pull request:
Resolves #4986 

## Description

> What is the purpose of this pull request?

This pull request:

-   {{TODO: add description describing what this pull request does}}

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #{{TODO: add issue number}} 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

shraddha761 committed:
bench: refactor random number stats/base/dists/signrank

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

Planeshifter reviewed (changes requested):
Thanks for your PR, @shraddha761! To complete the linked issue, please also make the following changes:

> Move random number generation out of the benchmarking loops and initialize it before the benchmarks.

See the linked PR for how this would look like.

shraddha761 committed:
fix: move random number generation outside benchmark loops

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

shraddha761 commented:
@Planeshifter did some changes as required

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/signrank/cdf/benchmark/benchmark.js:
```diff
@@ -31,16 +31,23 @@ var cdf = require( './../lib' );
 
 bench( pkg, function benchmark( b ) {
-	var n;
+	var nValues;
+	var values;
```
```suggestion
	var len;
	var n;
```

Please add only `len` and keep the original variable names unchanged (for consistency).

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/signrank/cdf/benchmark/benchmark.js:
```diff
@@ -31,16 +31,23 @@ var cdf = require( './../lib' );
+	var values;
 	var x;
-	var y;
 	var i;
+	var y;
```
```suggestion
	var y;
	var i;
```

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/signrank/cdf/benchmark/benchmark.js:
```diff
@@ -21,8 +21,8 @@
 var bench = require( '@stdlib/bench' );
-var randu = require( '@stdlib/random/base/randu' );
-var ceil = require( '@stdlib/math/base/special/ceil' );
+var uniform = require( '@stdlib/random/base/uniform' );
+var discreteUniform = require( '@stdlib/random/base/discrete-uniform' );
```
```suggestion
var Float64Array = require( '@stdlib/array/float64' );
var uniform = require( '@stdlib/random/base/uniform' );
var discreteUniform = require( '@stdlib/random/base/discrete-uniform' );
```

anandkaranubc replied:
`Float64Array` should be used instead of `Array`

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/signrank/cdf/benchmark/benchmark.js:
```diff
@@ -31,16 +31,23 @@ var cdf = require( './../lib' );
 	var i;
+	var y;
+
+	values = new Array( b.iterations );
+	nValues = new Array( b.iterations );
+	for ( i = 0; i < b.iterations; i++ ) {
+		values[ i ] = uniform( 0.0, 20.0 );
+		nValues[ i ] = discreteUniform( 1, 20 );
+	}
```
```suggestion
	len = 100;
	x = new Float64Array( len );
	n = new Float64Array( len );
	for ( i = 0; i < len; i++ ) {
		x[ i ] = uniform( 0.0, 20.0 );
		n[ i ] = discreteUniform( 1, 20 );
	}
```

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/signrank/cdf/benchmark/benchmark.js:
```diff
@@ -31,16 +31,23 @@ var cdf = require( './../lib' );
-		x = randu() * 20.0;
-		n = ceil( randu()*20.0 );
-		y = cdf( x, n );
+		x = values[ i ];
+		y = cdf( x, nValues[ i ] );
```
```suggestion
		y = cdf( x[ i % len ], n[ i % len ] );
```

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/signrank/cdf/benchmark/benchmark.js:
```diff
@@ -54,6 +61,7 @@ bench( pkg, function benchmark( b ) {
 });
 
 bench( pkg+':factory', function benchmark( b ) {
+	var values;
```
Same comment here and also applies throughout the PR.

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/signrank/cdf/benchmark/benchmark.js:
```diff
@@ -63,9 +71,14 @@ bench( pkg+':factory', function benchmark( b ) {
 	n = 20;
 	mycdf = cdf.factory( n );
 
+	values = new Array( b.iterations );
+	for ( i = 0; i < b.iterations; i++ ) {
+		values[ i ] = uniform( 0.0, 20.0 );
+	}
```
This should be updated too by following the example above.

anandkaranubc commented on lib/node_modules/@stdlib/stats/base/dists/signrank/quantile/benchmark/benchmark.js:
```diff
@@ -21,11 +21,12 @@
+var discreteUniform = require( '@stdlib/random/base/discrete-uniform' );
 var pkg = require( './../package.json' ).name;
 var quantile = require( './../lib' );
+var EPS = 1e-10;
```
`EPS` is not needed in this benchmark file, as the original code uses:  

```js
p = randu();
```  

Thus, using `p = uniform( 0.0, 1.0 );` is perfectly fine in this case.

anandkaranubc replied:
If it is ever used, it should be imported as:

```js
var EPS = require( '@stdlib/constants/float64/eps' );
```

anandkaranubc commented:
Thanks, @shraddha761, for working on this issue. I've left some feedback and an example for you to review. I hope that helps. You may also find the reference issues provided in https://github.com/stdlib-js/stdlib/issues/4986 useful too.

Can you also change the title to:

bench: refactor random number generation in `stats/base/dists/signrank`

Note: There are backticks around `stats/base/dists/signrank`
